### DISCUSSION TITLE
For Loops in Templates

### DISCUSSION THREAD

**Discussion by gruff101** [SilentVoid13/Templater#1133] [Help]
Hi,

Creating a template for the days of the week and want to generate the entries using a for loop with standard headings for each day.

Struggling to get this to work but assumed I could something along the lines below. But unable to get it to work. Can you help?

```
<%* 
for(i=0;i < 7;i++){
-%>

## Heading 1

## Heading 2

<%* 
}
-%>
```

**Comment by Zachatoo** [comment]
Appears to be working for me. Here's the result I got after applying the template. What happened for you? What did you expect to happen? How did you apply the template?

```


## Heading 1

## Heading 2


## Heading 1

## Heading 2


## Heading 1

## Heading 2


## Heading 1

## Heading 2


## Heading 1

## Heading 2


## Heading 1

## Heading 2


## Heading 1

## Heading 2


```

**Comment by gruff101** [comment]
Hi Zach,

I had a string of other code and when I bashed the code straight out it worked. 

What I don't understand is that the instance of "i" wasn't defined anywhere.

If i was doing it in other languages such as Java I would have had

```
<%* 
for(int i=0;i < 7;i++){
-%>

## Heading 1

## Heading 2

<%* 
}
-%>
```

If I'm doing this properly should I be defining i properly? Does it make any difference whether i is defined using var like below?
```
<%* 
for(var i=0;i < 7;i++){
-%>

## Heading 1

## Heading 2

<%* 
}
-%>
```

**Reply by Zachatoo** [reply]
You should define `i` with `var` (or even better, `let` or `const`).

Keep in mind that Templater is executing this code, so there are other variables that you can't see, you were likely updating some existing `i` variable in a higher scope that you can't see.

Or JS is just weird and lets you do that, but it's definitely not standard.
